src/commands/test/generate/uitest.ts

```diff
@@ -20,16 +20,25 @@ export default class GenerateUITestCommand extends GenerateCommand {
     const packageFilePath = path.join(this.outputPath, `AppCenter.UITest.${platform}/packages.config`);
     const projectFilePath = path.join(this.outputPath, `AppCenter.UITest.${platform}/AppCenter.UITest.${platform}.csproj`);
 
+    let latestVersion: string;
+
     try {
-      const latestVersion: string = await this.getLatestUITestVersionNumber();
+      latestVersion = await this.getLatestUITestVersionNumber();
+    } catch (e) {
+      console.warn("Can't retrieve latest UITest version. Using default version from template. Details: " + e);
+      return;
+    }
 
+    try {
       // Replace version inside packages.config file
       await this.replaceVersionInFile(packageFilePath, /(id="Xamarin\.UITest" version=")(\d+(\.\d+)+)/, latestVersion);
 
       // Replace version inside *.csproj file
       await this.replaceVersionInFile(projectFilePath, /(packages\\Xamarin\.UITest\.)(\d+(\.\d+)+)/, latestVersion);
     } catch (e) {
-      console.warn("Can't update UITest version. Using default version from template. Details: " + e);
+      await this.copyTemplates();
```

Comment:
Is it possible to add a few tests for this functionality? So we can verify that the `copyTemplates` function is called when `replaceVersionInFile` throws an error?

Reply:
Added two tests 😄